### ISSUE TITLE
Clang error when building with v1.4.0

### ISSUE THREAD

**Issue by jhuggart** [confluentinc/confluent-kafka-go#450]
Description
===========

I upgraded to the latest version of the go client (v1.4.0) and ran `brew upgrade librdkafka`. But the client is looking for librdkafka in the package directory and it is not present.

How to reproduce
================

In an existing project that uses confluent-kafka-go, run `go get -u gopkg.in/confluentinc/confluent-kafka-go.v1/kafka` and `go build ./...`

You should receive something similar to:

```
clang: error: no such file or directory: '$GOHOME/src/github.com/your-user/your-service/vendor/gopkg.in/confluentinc/confluent-kafka-go.v1/kafka/librdkafka
/librdkafka_darwin.a'
```

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-go and librdkafka version (`LibraryVersion()`): v1.4.0 (both)
 - [ n/a ] Apache Kafka broker version:
 - [ n/a ] Client configuration:
 - [x] Operating system: Mac OS X 10.15.4
 - [ n/a ] Provide client logs
 - [ n/a ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by jhuggart** [comment]
I fixed this by manually copying in the static `librdkafka` deps. But it looks like that dir is not making it into the published module.

**Comment by solarhell** [comment]
I ran into the same issue:

```
# github.com/confluentinc/confluent-kafka-go/kafka
vendor/github.com/confluentinc/confluent-kafka-go/kafka/00version.go:24:10: fatal error: 'librdkafka/rdkafka.h' file not found
#include <librdkafka/rdkafka.h>
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by edenhill** [comment]
My guess would be that you already have gopkg.in...v1 cached somewhere (to an older version, e.g. v1.3.0), try removing whatever confluent-kafka-go caches you find in $GOHOME and try again.

**Comment by solarhell** [comment]
after run `go clean -modcache` and `go mod tidy && go mod vendor`
i got this issue
`# github.com/confluentinc/confluent-kafka-go/kafka
clang: error: no such file or directory: '/Users/solarhell/go/src/github.com/solarhell/confluent-kafka-go-consumer/vendor/github.com/confluentinc/confluent-kafka-go/kafka/librdkafka/librdkafka_darwin.a'`

**Comment by edenhill** [comment]
can you do `ls -laR /Users/solarhell/go/src/github.com/solarhell/confluent-kafka-go-consumer/vendor/github.com/confluentinc/confluent-kafka-go/kafka/`  and provide the output?

**Comment by edenhill** [comment]
It looks like your vendor:ed confluent-kafka-go is missing the kafka/librdkafka directory. Make sure it is at version/tag v1.4.0

**Comment by solarhell** [comment]
i am sure i am using the 1.4.0 version.
![image](https://user-images.githubusercontent.com/10279583/78954596-2ee97880-7b0f-11ea-875e-3055a752fc75.png)

the output is
```
total 984
drwxr-xr-x  32 solarhell  staff    1024 Apr  9 19:41 .
drwxr-xr-x   4 solarhell  staff     128 Apr  9 19:41 ..
-rw-r--r--   1 solarhell  staff      50 Apr  9 19:41 .gitignore
-rw-r--r--   1 solarhell  staff    2087 Apr  9 19:41 00version.go
-rw-r--r--   1 solarhell  staff    3235 Apr  9 19:41 README.md
-rw-r--r--   1 solarhell  staff   34496 Apr  9 19:41 adminapi.go
-rw-r--r--   1 solarhell  staff    7221 Apr  9 19:41 adminoptions.go
-rw-r--r--   1 solarhell  staff  225603 Apr  9 19:41 api.html
-rw-r--r--   1 solarhell  staff     402 Apr  9 19:41 build_darwin.go
-rw-r--r--   1 solarhell  staff     210 Apr  9 19:41 build_dynamic.go
-rw-r--r--   1 solarhell  staff     420 Apr  9 19:41 build_glibc_linux.go
-rw-r--r--   1 solarhell  staff     432 Apr  9 19:41 build_musl_linux.go
-rw-r--r--   1 solarhell  staff    7705 Apr  9 19:41 config.go
-rw-r--r--   1 solarhell  staff   24171 Apr  9 19:41 consumer.go
-rw-r--r--   1 solarhell  staff     967 Apr  9 19:41 context.go
-rw-r--r--   1 solarhell  staff    4434 Apr  9 19:41 error.go
-rw-r--r--   1 solarhell  staff    2911 Apr  9 19:41 error_gen.go
-rw-r--r--   1 solarhell  staff   10185 Apr  9 19:41 event.go
-rw-r--r--   1 solarhell  staff   21804 Apr  9 19:41 generated_errors.go
-rw-r--r--   1 solarhell  staff    1180 Apr  9 19:41 glue_rdkafka.h
-rw-r--r--   1 solarhell  staff    9431 Apr  9 19:41 handle.go
-rw-r--r--   1 solarhell  staff    1942 Apr  9 19:41 header.go
-rw-r--r--   1 solarhell  staff   15294 Apr  9 19:41 kafka.go
-rw-r--r--   1 solarhell  staff    2240 Apr  9 19:41 log.go
-rw-r--r--   1 solarhell  staff    5844 Apr  9 19:41 message.go
-rw-r--r--   1 solarhell  staff    5121 Apr  9 19:41 metadata.go
-rw-r--r--   1 solarhell  staff     854 Apr  9 19:41 misc.go
-rw-r--r--   1 solarhell  staff    3965 Apr  9 19:41 offset.go
-rw-r--r--   1 solarhell  staff   30378 Apr  9 19:41 producer.go
-rw-r--r--   1 solarhell  staff     192 Apr  9 19:41 testconf-example.json
-rw-r--r--   1 solarhell  staff    5650 Apr  9 19:41 testhelpers.go
-rw-r--r--   1 solarhell  staff    1508 Apr  9 19:41 time.go
```

**Comment by edenhill** [comment]
It is missing the `librdkafka` directory, which I think is because it does not have any .go files in it.
But this seems to work for most people, so I wonder what is different with your setup / tooling.

**Comment by solarhell** [comment]
I can provide my dockerfile and sample app in a repo. Wait for a moment, pls.

**Comment by edenhill** [comment]
That'd be great, thank you!

**Comment by solarhell** [comment]
@edenhill https://github.com/solarhell/confluentKafkaGoExample

**Comment by edenhill** [comment]
Seems to be hitting this issue of go mod vendor:
https://github.com/golang/go/issues/26366

**Comment by edenhill** [comment]
Let's see if sticking a .go file in librdkafka/ solves the issue.

**Comment by solarhell** [comment]
Last time i ran into the similar issue and i used the 
```go
import (
	_ "mypackage"
)
``` solved the issue.

**Comment by edenhill** [comment]
@solarhell Can you try if a
```golang
import _ "librdkafka"
```

from e.g. `kafka/kafka.go`  fixes the problem on your end?

**Comment by solarhell** [comment]
It works!
But modify vendor is not a good way.

**Comment by solarhell** [comment]
The root problem is that the "librdkafka" package is not vendored.
So i can't import it.

**Comment by edenhill** [comment]
Can you try importing the vendor again using tag v1.0.7-RC1 (this is a dummy tag that will be removed, it is actually 1.4.0+)?

The librdkafka/ subdir should now be included.

Thanks

**Comment by solarhell** [comment]
It works!
![image](https://user-images.githubusercontent.com/10279583/79203097-1e1d6780-7e6d-11ea-8e17-ade410268b25.png)

**Comment by edenhill** [comment]
Whoop! 

Thank you so much for the quick testing!

I'll prepare a new release, v1.4.1, to fix this proper.

**Comment by solarhell** [comment]
Thanks for your work! 🎉

**Comment by edenhill** [comment]
Just to verify, the building and running of your application now works correctly with the bundled librdkafka, yes?

**Comment by solarhell** [comment]
Yes. I used the producer example to verify it on macOS.
![image](https://user-images.githubusercontent.com/10279583/79204014-799c2500-7e6e-11ea-8dea-2eaba4ace999.png)

**Comment by edenhill** [comment]
Perfect, thank you!

**Comment by blaketastic2** [comment]
Should I be using `v1.0.7-RC1`, or is there a `v1.4.1` version with this fix?

**Comment by jhuggart** [comment]
Any updates on this @edenhill ? I tried `go get gopkg.in/confluentinc/confluent-kafka-go.v1@v1.0.7-RC1` but get `invalid version: unknown revision v1.0.7-RC1`.

**Comment by fnuneztw** [comment]
I can't find any `v1.0.7-RC1` tag to use as a temporary fix as mentioned [here](https://github.com/confluentinc/confluent-kafka-go/issues/450#issuecomment-613300087). Was it already removed?

**Comment by edenhill** [comment]
Yes, it was removed, we're aiming to release a proper v1.4.2 this week, will have a release-candidate ready soon for you to try out.

**Comment by edenhill** [comment]
Verified on `golang:1.13.10-alpine3.10` (docker):

```
librdkafka link information: static musl_linux from librdkafka-static-bundle-v1.4.2-RC1.tgz
%7|1587992099.237|OPENSSL|rdkafka#producer-1| [thrd:app]: librdkafka built with statically linked OpenSSL version 0x1000215f

%7|1587992099.237|CACERTS|rdkafka#producer-1| [thrd:app]: Setting default CA certificate location to /etc/ssl/cert.pem, override with ssl.ca.location

^^^^ this is the probed location

%7|1587992099.249|INIT|rdkafka#producer-1| [thrd:app]: librdkafka v1.4.2-RC1 (0x10402ff) rdkafka#producer-1 initialized (builtin.features gzip,snappy,ssl,sasl,regex,lz4,sasl_plain,sasl_scram,plugins,zstd,sasl_oauthbearer, STATIC_LINKING GCC GXX PKGCONFIG GNULD LDS C11THREADS LIBDL PLUGINS STATIC_LIB_zlib ZLIB STATIC_LIB_libcrypto STATIC_LIB_libssl SSL STATIC_LIB_libzstd ZSTD HDRHISTOGRAM SYSLOG SNAPPY SOCKEM SASL_SCRAM SASL_OAUTHBEARER CRC32C_HW, debug 0x200)

%7|1587992099.391|SSLVERIFY|rdkafka#producer-1| [thrd:sasl_ssl://.....confluent.cloud:9092/boot]: sasl_ssl://....confluent.cloud:9092/bootstrap: Broker SSL certificate verified
```

**Comment by fnuneztw** [comment]
I guess now I can use `librdkafka v1.4.2-RC1` as a temporary fix, right?

**Comment by edenhill** [comment]
librdkafka still needs to be bundled with the Go client.
To keep the Go repo size down we'll only do this once librdkafka v1.4.2 is released which is hopefully later this week.

**Comment by edenhill** [comment]
The workaround for Alpine is to set `"ssl.ca.location": "/etc/ssl/cert.pem"` in the ConfigMap

**Comment by daniesy** [comment]
Any news on the release? Or a temporary fix?

**Comment by edenhill** [comment]
Fixed in v1.4.2

**Comment by kutty-kumar** [comment]
@edenhill this isn't working using dep

**Comment by edenhill** [comment]
@kutty-kumar Please provide more information. Steps to reproduce, error messages, file listings,  mod files, etc

**Comment by mostafatalebi** [comment]
The problem is fixed with v1.4.2. But we should import 
`_ "gopkg.in/confluentinc/confluent-kafka-go.v1/kafka/librdkafka"`

Thanks!
